fealib: allow newlines in string literals (and skip over them)

anthrotype opened the pull request:
should fix #578 

/cc @brawer

brawer commented on Lib/fontTools/feaLib/lexer.py:
```diff
@@ -142,13 +142,24 @@ def next_(self):
-                                      location)
+                if cur_char == '"':
+                    strings.append(text[prev_pos:self.pos_ - 1])
+                    return (Lexer.STRING, "".join(strings), location)
```
Wouldn’t it be easier to do self.scan_util('"') and then replace \r and \n by '' ?

anthrotype replied:
I can have a look tomorrow. If you want, you can go on and fix it yourself, no prob ;)

anthrotype commented:
@brawer I simplified the code a bit, like you suggested, and added a test case. It should be good to go now. Feel free to add to it in case I missed something.

brawer commented:
👍
